src/traces/funnelarea/attributes.js

```diff
@@ -0,0 +1,85 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var pieAttrs = require('../pie/attributes');
+var sunburstMarker = require('../sunburst/attributes').marker;
+var plotAttrs = require('../../plots/attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var hovertemplateAttrs = require('../../components/fx/hovertemplate_attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+    labels: pieAttrs.labels,
+    // equivalent of x0 and dx, if label is missing
+    label0: pieAttrs.label0,
+    dlabel: pieAttrs.dlabel,
+    values: pieAttrs.values,
+    marker: sunburstMarker,
+    text: pieAttrs.text,
+    hovertext: pieAttrs.hovertext,
+
+    scalegroup: extendFlat({}, pieAttrs.scalegroup, {
+        description: [
+            'If there are multiple funnelareas that should be sized according to',
+            'their totals, link them by providing a non-empty group id here',
+            'shared by every trace in the same group.'
+        ].join(' ')
+    }),
+
+    textinfo: extendFlat({}, pieAttrs.textinfo, {
+        flags: ['label', 'text', 'value', 'percent']
+    }),
+
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['label', 'text', 'value', 'percent', 'name']
+    }),
+
+    hovertemplate: hovertemplateAttrs({}, {
+        keys: ['label', 'color', 'value', 'percent', 'text']
+    }),
+
+    textfont: pieAttrs.textfont,
+    insidetextfont: pieAttrs.insidetextfont,
+    outsidetextfont: pieAttrs.outsidetextfont,
+
+    title: extendFlat({}, pieAttrs.title, {
+        position: extendFlat({}, pieAttrs.title.position, {
+            values: ['top left', 'top center', 'top right']
+        })
+    }),
+
+    domain: domainAttrs({name: 'funnelarea', trace: true, editType: 'calc'}),
+
+    aspectratio: {
+        valType: 'number',
+        role: 'info',
+        min: 0,
+        dflt: 1,
+        editType: 'plot',
+        description: [
+            'Sets the ratio between height and width'
+        ].join(' ')
+    },
+
+    baseratio: {
+        valType: 'number',
+        role: 'info',
+        min: 0,
+        max: 1,
+        dflt: 0.333,
+        editType: 'plot',
+        description: [
+            'Sets the ratio between bottom length and maximum top length.'
+        ].join(' ')
+    },
+
+    sort: extendFlat({}, pieAttrs.sort, { dflt: false })
```

Comment:
I would :hocho: this, to leave room for improvement for future iterations. @archmoj what made you choose to add this in afterall?

Reply:
1. Aggregated funnelarea graphs are not looking great without sorting.
2. In case of having a sorted pie with not sorted funnelarea labels may strangely hide other colors.

Reply:
> 1. Aggregated funnelarea graphs are not looking great without sorting.

Ok, but if `sort` defaults to false, then "Aggregated funnelarea graphs" will still "not look great" by default.

> In case of having a sorted pie with not sorted funnelarea labels may strangely hide other colors.

Can you share an example of this case?

Reply:
issue 1: [Codepen](https://codepen.io/MojtabaSamimi/pen/gJXmqE?editors=0010);
issue 2: [Codepen](https://codepen.io/MojtabaSamimi/pen/zQPZzK?editors=0010).

Reply:
You're right @archmoj - `funnelarea` traces that have aggregated values look a lot better sorted AND are a lot more useful.

So I think we should still :hocho: to `sort` attribute, but behind the scenes sort aggregated funnelarea calcdata during the calc step. I can't think of a reason to _not_ sort aggragated `funnelarea` traces.

Reply:
Issue 1 and Issue 2 have been resolved in the scope of this PR.